### PR TITLE
Search CRUD topics

### PR DESCRIPTION
**Pull request by jamesm-macrometa**
This PR adds API, Python, JS, and CLI instructions for creating a Search View.

Reviewers:  Please take a look at the JavaScript example.  It seems to be in the right direction, and it runs without errors, but for some reason it does not actually create a Search View like the Python example does.  ChatGPT has run out of ideas troubleshooting and suggested I contact Macrometa support.

### CODE REVIEW

File: docs/search/views/create-search-views.md [hunk 1]
```diff
@@ -24,3 +30,145 @@ Follow these instructions to create a new search view using the GDN console web
...
+  
+  # Create test view on demo collection with store-values
+  gdnsl view create test --collection demo --store-values none
+
+  # Create test view on demo collection with track-list-positions
+  gdnsl view create test --collection demo --track-list-positions
+
+  # Create test view on demo collection with include-all-fields
+  gdnsl view create test --collection demo --include-all-fields
+
```
Comment (koshy-thomas) [inline docs/search/views/create-search-views.md]: we should also add gdnsl example command with primary sort field..

gdnsl view create test --collection demo --field "name:text_en,text_fr" --field "age:text_en" --primary-sort-field "name:asc"

File: docs/search/views/create-search-views.md [hunk 2]
```diff
@@ -24,3 +30,145 @@ Follow these instructions to create a new search view using the GDN console web
...
+      --analyzer string        Analyzers to be used for indexing of string values. Can be given multiple times. (default: identity).
+      --field string           Field and analyzers mapping in the format <fieldName:analyzer1,analyzer2> 
+                               Example: "age:text_en,text_fr". Can be given multiple times.
+      --store-values string    How should the view track the attribute values, this setting allows for additional value retrieval optimizations, one of:
+                               none: Do not store values by the view.
+                               id: Store only information about value presence, to allow use of the EXISTS() function (default "none").
+      --track-list-positions boolean  The flag determines whether or not values in a lists should be treated separate (default: false).
+      --include-all-fields boolean    The flag determines whether or not to index all fields on a particular level of depth (default: false).
+      --fabric                 Name of the fabric to use.
+```
```
Comment (koshy-thomas) [inline docs/search/views/create-search-views.md]: --primary-sort-field  string  The flag accepts the fields and the sort order by which the collection will be sorted. Format "fieldName:sortOrder (either "asc" or "desc")".
Example: --primary-sort-field "name:asc".

File: docs/search/views/create-search-views.md [hunk 3]
```diff
@@ -24,3 +30,145 @@ Follow these instructions to create a new search view using the GDN console web
...
+# Choose one of the following methods to access the GDN. API key is recommended.
+
+# Authenticate with API key.
+client = C8Client(protocol='https', host=URL, port=443, apikey=API_KEY, geofabric=GEO_FABRIC)
+
+
+search_view_name = "example_search_view"
+collection_name = "your_collection_name"
+properties = {
+    "<COLLECTION NAME>": { # Change this to the name of the collection to use with the search view.
```
Comment (koshy-thomas) [inline docs/search/views/create-search-views.md]: collection_name is already wrapped we need to remove this extra wrapping.. let's have a call @jamesm-macrometa when you are online we will also add primary sort example here too and the same for Javascript too and will give you the working pyC8, jsc8 codes

File: docs/search/views/create-search-views.md (hunk 1 above)
Comment (diana-macrometa) [reply docs/search/views/create-search-views.md]: I generally just link to the CLI topic or heading with the info needed, cuts down on the maintenance overhead. If this is wrong, though, we should also review the source CLI topic, @jamesm-macrometa .

**Comment by jamesm-macrometa** [comment]
@brandonLyonMM @diana-macrometa Any ideas why this branch is failing to build?  When I run the test on my own machine, it builds without any issues.

**Comment by diana-macrometa** [comment]
@jamesm-macrometa, did you do `yarn run build` to check for broken links?

**Comment by brandonLyonMM** [comment]
> @brandonLyonMM @diana-macrometa Any ideas why this branch is failing to build? When I run the test on my own machine, it builds without any issues.

The error message from our CI-CD builder is:

> Error: Docs markdown link couldn't be resolved: (../../CLI/search-views-cli.md) in "/vercel/path0/docs/search/views/create-search-views.md" for version current

**Comment by diana-macrometa** [comment]
@brandonLyonMM, I actually cannot see the Vercel error messages.

**Comment by jamesm-macrometa** [comment]
@brandonLyonMM @diana-macrometa This is very strange, the link works perfectly fine for me in VSCode and does not return any error when I run `yarn run build` on my system.  Any ideas how to fix this?

**Comment by diana-macrometa** [comment]
Are you sure you are on the correct, updated branch? Because I got an error on that branch as well.

**Comment by jamesm-macrometa** [comment]
> Are you sure you are on the correct, updated branch? Because I got an error on that branch as well.

I think my issues were caused by a broken link in Main, opened https://github.com/Macrometacorp/docs/pull/842 to fix it.

**Comment by jamesm-macrometa** [comment]
@brandonLyonMM @diana-macrometa For some reason the link I added didn't return an error for my local build, but when I removed `.md` from the link it works now.  Thanks for the help!

**Review by diana-macrometa — APPROVED** [review]

**Comment by jamesm-macrometa** [comment]
Created https://github.com/Macrometacorp/docs/pull/844/files to address Koshy's comments re: the CLI commands.

**Pull request merged**